### PR TITLE
iOS XR/XE SNMP Updates

### PR DESCRIPTION
**Pull request by mwdean**
Updated the XR snmp file with setting community and access, along with an accompanying XE file.

### CODE REVIEW

File: Cisco/iOS-XR/snmp.conf [hunk 1]
```diff
@@ -1,2 +1,10 @@
-! making sure ifIndex gets persisted with SNMP after router reboot 
+! SNMP IOS-XR
+!
+! Set SNMP Access ACL
+ipv4 access-list 33
+32 remark NNS SNMP Access
```
Comment (grrrrreg) [inline Cisco/iOS-XR/snmp.conf]: I believe it should rather be:
`33 remark NNS SNMP Access `
given that the ACL below says 33 ?

Also same comment that for iOS-XE config RE: remark statement:
`access-list 33 remark Kentik SNMP Polling ACL US and EU`

File: Cisco/iOS-XE/snmp.conf [hunk 2]
```diff
@@ -0,0 +1,9 @@
+! SNMP IOS-XE
+!
+! ACL for SNMP Access
+access-list 33 remark SNMP Query ACL
```
Comment (grrrrreg) [inline Cisco/iOS-XE/snmp.conf]: Suggested change (see next comment in review for US and EU clarification)
`access-list 33 remark Kentik SNMP Polling ACL US and EU`

File: Cisco/iOS-XE/snmp.conf [hunk 3]
```diff
@@ -0,0 +1,9 @@
...
+access-list 33 permit 209.50.158.0 0.0.1.255
```
Comment (grrrrreg) [inline Cisco/iOS-XE/snmp.conf]: Would add a line to permit polling from the EU platform:
`access-list 33 permit 193.177.128.0 0.0.3.255`
and users would choose to keep one or both

File: Cisco/iOS-XR/snmp.conf [hunk 4]
```diff
@@ -1,2 +1,10 @@
...
+33 permit ipv4 209.50.158.0 0.0.1.255 any
+! Set SNMP Acess and Community
+snmp-server community <SNMP_COMMUNITY> RO IPv4 33
```
Comment (grrrrreg) [inline Cisco/iOS-XR/snmp.conf]: In this line, using `{{snmp_community}}` will leverage templating and auto-replace with what is inserted in the portal UI at device provisioning time - this isn't leveraged in the current UI, but will be in the forthcoming device provisioning UI which will suggest configuration snippets directly related to the values entered in the device provisioning web form. (the Juniper templates already leverage that `{{}}` notation.

File: Cisco/iOS-XE/snmp.conf [hunk 5]
```diff
@@ -0,0 +1,9 @@
...
+!
+! Sets Access and the community
+snmp-server community <SNMP_COMMUNITY> RO 33
```
Comment (grrrrreg) [inline Cisco/iOS-XE/snmp.conf]: In this line, using `{{snmp_community}}` will leverage templating and auto-replace with what is inserted in the portal UI at device provisioning time - this isn't leveraged in the current UI, but will be in the forthcoming device provisioning UI which will suggest configuration snippets directly related to the values entered in the device provisioning web form. (the Juniper templates already leverage that `{{}}` notation.

File: Cisco/iOS-XR/snmp.conf (hunk 1 above)
Comment (mwdean) [reply Cisco/iOS-XR/snmp.conf]: Indeed!

File: Cisco/iOS-XE/snmp.conf (hunk 3 above)
Comment (mwdean) [reply Cisco/iOS-XE/snmp.conf]: Will do

File: Cisco/iOS-XR/snmp.conf (hunk 4 above)
Comment (mwdean) [reply Cisco/iOS-XR/snmp.conf]: Oh wow, cool. I was wondering why you all chose to use double {{}}.

Will do!

File: Cisco/iOS-XE/snmp.conf (hunk 5 above)
Comment (mwdean) [reply Cisco/iOS-XE/snmp.conf]: yes!

**Comment by mwdean** [comment]
I have made all the suggested changes!

Thanks Greg!

**Pull request closed without merging**